cpp/src/gandiva/function_registry_string.cc

```diff
@@ -114,6 +114,22 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
                      utf8(), kResultNullIfNull, "substr_utf8_int64_int64",
                      NativeFunction::kNeedsContext),
 
+      NativeFunction("to_char", {}, DataTypeVector{int32(), int64()}, utf8(),
```

Comment:
The signature looks incorrect. Aren't you taking pattern (string) as second argument.
It should be {int32(), utf8()}